### PR TITLE
Redecorate with dark theme

### PR DESCRIPTION
**Pull request by pinheadmz**
Offering this for late night review club crammers to make the site easier on the eyes. For IRC logs, I switched out a few colors for better contrast against the dark background.

Here's some examples screen shots showing main page, IRC log, and a `code` block...

##

Screen Shot 2021-02-17 at 9 59 48 AM

##

Screen Shot 2021-02-17 at 10 00 00 AM

##

Screen Shot 2021-02-17 at 10 01 33 AM

### CODE REVIEW

**Comment by pinheadmz** [comment]
Converting to draft for now, will investigate the simplest way to make this user settable. @jonatack @jnewbery this may introduce javascript to the website, is that a game changer?

**Comment by jnewbery** [comment]
> Converting to draft for now, will investigate the simplest way to make this user settable. @jonatack @jnewbery this may introduce javascript to the website, is that a game changer?

I'd definitely prefer it if there was a way to achieve this without adding javascript.

**Comment by jonatack** [comment]
> I'd definitely prefer it if there was a way to achieve this without adding javascript.

Same, though a quick web search for doing this with jekyll turns up solutions that need a tiny bit of JS to get around the static aspect (often with media queries and localStorage). Maybe with this: https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

**Comment by jnewbery** [comment]
@pinheadmz are you still working on this?

**Comment by pinheadmz** [comment]
Well, I haven't given up on this, but you can close if you are just cleaning up the repository. I discovered a trick using HTML and CSS they can actually switch some color variables in other areas of the style. But haven't got it quite to work how I envisioned it yet.

**Comment by jnewbery** [comment]
No need to close this. I just wanted to check that it was still active.

**Comment by pinheadmz** [comment]
Ok, refactored this with a toggle button (💡 emoji floating in top-right corner) and the actual CSS color switching requires NO javascript. However -- getting the user preference to persist does. Users with JS disabled will have to click the button every time they switch pages or reload. All I'm doing in the JS is saving a boolean in the browser local storage.


Mar-30-2021 17-01-06

**Comment by jnewbery** [comment]
Very cool. This looks good to me. Thanks @pinheadmz!

@jonatack - you're a dark mode connoisseur. Could you check that this satisfies your needs (and merge if so)?

**Comment by jonatack** [comment]
Thanks @pinheadmz! Using the Dark Reader extension slows things down and is a privacy hole, so this is great.

Tested and the only issues I see are

- White background at the bottom of the page (see screenshot below)
- Clicked links don't seem to be changing color

Do you want to have a look at these? Otherwise this is :+1:

**Comment by jonatack** [comment]
(tested with the Firefox 87 on Linux)

Screenshot from 2021-03-31 12-14-49

**Comment by pinheadmz** [comment]
@jonatack thanks for taking a look. I think I figured out the nits. Tested on OSX Chrome / Safari / Firefox and iOS Chrome

File: assets/css/all.sass [hunk 1]
```diff
@@ -59,19 +69,26 @@ a
 a
   &:hover
     color: $secondary-color
+a
+  &:visited
+    color: var(--link-visited-color)
```
Comment (jonatack) [inline assets/css/all.sass:74]: By inverting these two, ISTM we get the hover action on the visited links just like on the not-yet visited ones.
```diff
-a
-  &:hover
-    color: $secondary-color
 a
   &:visited
     color: var(--link-visited-color)
 
+a
+  &:hover
+    color: $secondary-color
+
```

**Review by jonatack — COMMENTED** [review]
This is great--thanks for keeping at it! I tested again on Firefox and compared to the production site without and without the dark reader extension. LGTM, only one remaining suggestion.

File: assets/css/all.sass (hunk 1 above)
Comment (pinheadmz) [reply assets/css/all.sass:71]: Great idea, added in 1b7d6d520ac056a947a66135aa3bfd979ecdcdce

tbh, I didn't even realize the `.sass` files were top-down hierarchical 🤦

**Comment by jonatack** [comment]
Tested ACK.

The remaining things I looked at:
- The color of the log line links in default (light) mode has changed from grey to the blue for links. I liked the grey but it seems to be more coherent now
- The blue color of the unvisited links is a bit darker now in default (light mode). I played with it briefly and didn't see why.
- The colors in dark mode are really dark compared to dark reader (but it's just habit, not an issue)
- I'm not sure the meaning of the light bulb icon is clear to people on its own. Maybe a "dark mode/light mode" link to the right of the code of conduct one.

If any of these are a concern, they can be done in a follow-up.

Thanks @pinheadmz!

**Comment by jnewbery** [comment]
Thanks zippy!

**Pull request merged**